frontend/js/src/pins/PinRecordingModal.tsx

```diff
@@ -28,8 +29,13 @@ export const maxBlurbContentLength = 280;
  */
 
 export default NiceModal.create(
-  ({ recordingToPin, onSuccessfulPin }: PinRecordingModalProps) => {
+  ({
+    recordingToPin,
+    onSuccessfulPin,
+    rowId,
+  }: PinRecordingModalProps) => {
     const modal = useModal();
+    const isUpdate = Boolean(rowId);
     const [blurbContent, setBlurbContent] = React.useState("");
```

Comment:
Set initial blurb content here, fallback to the empty string as before:
```suggestion
    const [blurbContent, setBlurbContent] = React.useState(initialBlurbContent ?? "");
```

With the other related changes that should mean you have the existing content in the textarea when you open the modal